src/util-inl.h

```diff
@@ -229,30 +229,74 @@ bool StringEqualNoCaseN(const char* a, const char* b, size_t length) {
   return true;
 }
 
+template <typename T>
+T MultiplyWithOverflowCheck(T a, T b) {
```

Comment:
>I can add a static_assert to check for unsignedness, if you want.

That works, but I would personally prefer to have a function that works just with `size_t`. That way you can see that it doesn't work with signed values from the signature. Your call, though.